### PR TITLE
fix: various fixes to enable searching for mappings

### PR DESCRIPTION
**Pull request by Ben-Sheppard**
## Description

As part of a separate piece of work I noticed that there were some things missing related to the mapping entity, this PR adds in the missing classes such that mappings can be retrieved from ES/OS.

### CODE REVIEW

File: search/search-domain/src/main/java/io/camunda/search/filter/MappingFilter.java [hunk 1]
```diff
@@ -7,13 +7,18 @@
...
 package io.camunda.search.filter;
 
+import static io.camunda.util.CollectionUtil.addValuesToList;
+
 import io.camunda.util.ObjectBuilder;
+import java.util.List;
 
-public record MappingFilter(Long mappingKey, String claimName, String claimValue)
+public record MappingFilter(
+    Long mappingKey, String claimName, List<String> claimNames, String claimValue)
```
Comment (emulbreh) [inline search/search-domain/src/main/java/io/camunda/search/filter/MappingFilter.java:16]: Can we replace the singular `claimName` filter with a single `claimNames` list filter? The API would be a bit simpler if there's only one way to express a filter.

File: search/search-client-query-transformer/src/test/java/io/camunda/search/clients/transformers/filter/MappingQueryTransformerTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    }
+  }
+
+  public static Stream<Arguments> queryFilterParameters() {
+    return Stream.of(
+        Arguments.of(
+            (Function<Builder, ObjectBuilder<MappingFilter>>)
+                f -> f.claimNames(List.of("foo", "bar")),
+            "claimName",
+            List.of("foo", "bar")),
```
Comment (emulbreh) [inline search/search-client-query-transformer/src/test/java/io/camunda/search/clients/transformers/filter/MappingQueryTransformerTest.java:63]: If we construct a full `SearchQuery` as the expected value here we can get rid of the instanceof checks and also make it a bit more obvious what we're actually testing. 
```
SearchQuery.of(q -> q.terms(t -> t.stringTerms(List.of("foo", "bar")))
```

File: search/search-domain/src/main/java/io/camunda/search/filter/MappingFilter.java (hunk 1 above)
Comment (Ben-Sheppard) [reply search/search-domain/src/main/java/io/camunda/search/filter/MappingFilter.java:16]: We could but for now I'd prefer to keep it this way, its a pattern that we use consistently in other filters where there could be one or multiple keys i.e. AuthorizationFilter.

File: search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/MappingFilterTransformer.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import io.camunda.search.clients.query.SearchQuery;
+import io.camunda.search.filter.MappingFilter;
+import java.util.List;
+
+public class MappingFilterTransformer implements FilterTransformer<MappingFilter> {
+  @Override
+  public SearchQuery toSearchQuery(final MappingFilter filter) {
+    return and(
+        stringTerms(CLAIM_NAME, filter.claimNames()),
```
[the remark this answers is not in the document]
Comment (Ben-Sheppard) [reply search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/MappingFilterTransformer.java:24]: The `stringTerms` method here caters for null, if `filter.claimNames()` is null then no filters will be added :)

[the remark this answers is not in the document]
Comment (emulbreh) [reply search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/MappingFilterTransformer.java:24]: Ah, that's interesting, maybe we could do the same for the single term filter so all these null checks can go away.

File: search/search-client-query-transformer/src/test/java/io/camunda/search/clients/transformers/filter/MappingQueryTransformerTest.java (hunk 2 above)
Comment (Ben-Sheppard) [reply search/search-client-query-transformer/src/test/java/io/camunda/search/clients/transformers/filter/MappingQueryTransformerTest.java:63]: Great suggestion, I've changed this now

File: search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/MappingFilterTransformer.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ben-Sheppard) [reply search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/MappingFilterTransformer.java:24]: Yea that could be an idea, though I'd defer to another PR for that as it has wider considerations I think :)

**Review by emulbreh — APPROVED** [review]

**Review by Oleksiivanov — APPROVED** [review]

**Pull request merged**